File: Homework/Mary_Antonyan/C++/30_03_2017/Chess/board.cpp
#include "board.hpp"
#include "figure.hpp"
#include "king.hpp"
#include "queen.hpp"
#include "knight.hpp"
#include "rook.hpp"
#include "bishop.hpp"
#include "pawn.hpp"
#include <iostream>

void Chess::initBoard() {
    Chess::board = new Chess::Figure**[8];
    for (int i = 0; i < 8; ++i) {
        Chess::board[i] = new Chess::Figure*[8];
    }

    for (int j = 0; j < 8; ++j) {
        Chess::board[1][j] = new Chess::Pawn(Chess::Figure::Chessman::pawn, Chess::Figure::Color::black, 1, j);
        Chess::board[6][j] = new Chess::Pawn(Chess::Figure::Chessman::pawn, Chess::Figure::Color::white, 6, j);
    }

    Chess::Figure::Color color = Chess::Figure::Color::black;
    for (int i = 0; i < 8; ++i) {
        for (int j = 0; j < 8; ++j) {
            if (0 == i || 7 == i) {
                switch (j) {
                    case 0 : case 7 : { 
                                          Chess::board[i][j] = new Chess::Rook(Chess::Figure::Chessman::rook, color, i, j);
                                          break;
                                      } 
                    case 1 : case 6 : { 
                                          Chess::board[i][j] = new Chess::Knight(Chess::Figure::Chessman::knight, color, i, j);
                                          break;
                                      }
                    case 2 : case 5 : { 
                                          Chess::board[i][j] = new Chess::Bishop(Chess::Figure::Chessman::bishop, color, i, j);
                                          break;
                                      }

                    case 3 : { 
                                 Chess::board[i][j] = new Chess::Queen(Chess::Figure::Chessman::queen, color, i, j);
                                 break;
                             }
                    case 4 : { 
                                 Chess::board[i][j] = new Chess::King(Chess::Figure::Chessman::king, color, i, j);
                                 break;
                             }
                }
            } else if (1 != i && 6 != i) {
                Chess::board[i][j] = nullptr;
            }
        }
        color = Chess::Figure::Color::white;
    }
}

void Chess::printBoard() {
    for (int i = 0; i < 8; ++i) {
        for (int j = 0; j < 8; ++j) {
            std::cout << "+----";
        }
        std::cout << "+" << std::endl << "|";

        for (int j = 0; j < 8; ++j) {
            if (nullptr == Chess::board[i][j]) {
                std::cout << "    |";
            } else {
                switch (Chess::board[i][j]->getFigure()) {
                    case Chess::Figure::Chessman::king : {
                                                             if (Chess::board[i][j]->getColor() == Chess::Figure::Color::white) {
                                                                 std::cout << " WK |";
                                                             } else if (Chess::board[i][j]->getColor() == Chess::Figure::Color::black) {
                                                                 std::cout << " BK |";
                                                             }
                                                             break;
                                                         }
                    case Chess::Figure::Chessman::queen : {
                                                              if (Chess::board[i][j]->getColor() == Chess::Figure::Color::white) {
                                                                  std::cout << " WQ |";
                                                              } else if (Chess::board[i][j]->getColor() == Chess::Figure::Color::black) {
                                                                  std::cout << " BQ |";
                                                              }
                                                              break;
                                                          }
                    case Chess::Figure::Chessman::knight : {
                                                               if (Chess::board[i][j]->getColor() == Chess::Figure::Color::white) {
                                                                   std::cout << " Wk |";
                                                               } else if (Chess::board[i][j]->getColor() == Chess::Figure::Color::black) {
                                                                   std::cout << " Bk |";
                                                               }
                                                               break;
                                                           }
                    case Chess::Figure::Chessman::rook : {
                                                             if (Chess::board[i][j]->getColor() == Chess::Figure::Color::white) {
                                                                 std::cout << " WR |";
                                                             } else if (Chess::board[i][j]->getColor() == Chess::Figure::Color::black) {
                                                                 std::cout << " BR |";
                                                             }
                                                             break;
                                                         }
                    case Chess::Figure::Chessman::bishop : {
                                                               if (Chess::board[i][j]->getColor() == Chess::Figure::Color::white) {
                                                                   std::cout << " WB |";
                                                               } else if (Chess::board[i][j]->getColor() == Chess::Figure::Color::black) {
                                                                   std::cout << " BB |";
                                                               }
                                                               break;
                                                           }
                    case Chess::Figure::Chessman::pawn : {
                                                             if (Chess::board[i][j]->getColor() == Chess::Figure::Color::white) {
                                                                 std::cout << " WP |";
                                                             } else if (Chess::board[i][j]->getColor() == Chess::Figure::Color::black) {
                                                                 std::cout << " BP |";
                                                             }
                                                             break;
                                                         }
                }
            }
        }
        std::cout << std::endl;
    }

    for (int j = 0; j < 8; ++j) {
        std::cout << "+----"; 
    }
    std::cout << "+" << std::endl;
}

void Chess::execute(const int fromRow, const int fromCol, 
        const int toRow, const int toCol) {
    if (nullptr != Chess::board[fromRow][fromCol]) {
        Chess::board[fromRow][fromCol]->step(toRow, toCol);
        Chess::printBoard();
    } else {
        std::cout << "Node is empty" << std::endl;
    }
}

File: Homework/Mary_Antonyan/Java/07_08_2017/Chess/src/main/java/com/itc/chess/Board.java
package com.itc.chess;
import java.awt.Point;

/**
 * This class represents the game board(8x8)
 * 
 * @author mary
 */
public class Board {
    public static Figure[][] board;

    /**
     * Initialized board content
     */
    public static void initBoard() {
        board = new Figure[8][8];
        for (int j = 0; j < 8; ++j) {
            board[1][j] = new Pawn(Figure.Chessman.pawn, Figure.Color.black, new Point(1, j));
            board[6][j] = new Pawn(Figure.Chessman.pawn, Figure.Color.white, new Point(6, j));
        }

        Figure.Color color = Figure.Color.black;
        for (int i = 0; i < 8; ++i) {
            for (int j = 0; j < 8; ++j) {
                if (0 == i || 7 == i) {
                    switch (j) {
                        case 0 : case 7 : { 
                            board[i][j] = new Rook(Figure.Chessman.rook, color, new Point(i, j));
                            break;
                        } 
                        case 1 : case 6 : { 
                            board[i][j] = new Knight(Figure.Chessman.knight, color, new Point(i, j));
                            break;
                        }
                        case 2 : case 5 : { 
                            board[i][j] = new Bishop(Figure.Chessman.bishop, color, new Point(i, j));
                            break;
                        }
                        case 3 : { 
                            board[i][j] = new Queen(Figure.Chessman.queen, color, new Point(i, j));
                            break;
                        }
                        case 4 : { 
                            board[i][j] = new King(Figure.Chessman.king, color, new Point(i, j));
                            break;
                        }
                    }
                } else if (1 != i && 6 != i) {
                    board[i][j] = null;
                }
            }
            color = Figure.Color.white;
        }
    }
    
    /**
     * Prints board content
     */
    public static void printBoard() {
        for (int i = 0; i < 8; ++i) {
            for (int j = 0; j < 8; ++j) {
                System.out.print("+----");
            }
            System.out.println("+"); 
            System.out.print("|");

            for (int j = 0; j < 8; ++j) {
                if (board[i][j] == null) {
                    System.out.print("    |");
                } else {
                    switch (board[i][j].getFigure()) {
                        case king : {
                            if (board[i][j].getColor() == Figure.Color.white) {
                                System.out.print(" WK |");
                            } else if (board[i][j].getColor() == Figure.Color.black) {
                                System.out.print(" BK |");
                            }
                            break;
                        }
                        case queen : {
                            if (board[i][j].getColor() == Figure.Color.white) {
                                System.out.print(" WQ |");
                            } else if (board[i][j].getColor() == Figure.Color.black) {
                                System.out.print(" BQ |");
                            }
                            break;
                        }
                        case knight : {
                            if (board[i][j].getColor() == Figure.Color.white) {
                                System.out.print(" Wk |");
                            } else if (board[i][j].getColor() == Figure.Color.black) {
                                System.out.print(" Bk |");
                            }
                            break;
                        }
                        case rook : {
                            if (board[i][j].getColor() == Figure.Color.white) {
                                System.out.print(" WR |");
                            } else if (board[i][j].getColor() == Figure.Color.black) {
                                System.out.print(" BR |");
                            }
                            break;
                        }
                        case bishop : {
                            if (board[i][j].getColor() == Figure.Color.white) {
                                System.out.print(" WB |");
                            } else if (board[i][j].getColor() == Figure.Color.black) {
                                System.out.print(" BB |");
                            }
                            break;
                        }
                        case pawn : {
                            if (board[i][j].getColor() == Figure.Color.white) {
                                System.out.print(" WP |");
                            } else if (board[i][j].getColor() == Figure.Color.black) {
                                System.out.print(" BP |");
                            }
                            break;
                        }
                    }
                }
            }
            System.out.println();

            for (int j = 0; j < 8; ++j) {
                System.out.print("+----"); 
            }
            System.out.println("+");
        }
    }

    /**
     * Executes the game
     * 
     * @param from object initial position
     * @param to object new position
     */
    public static void execute(Point from, Point to) {
        if (board[from.x][from.y] != null) {
            board[from.x][from.y].step(to);
            printBoard();
        } else {
            System.out.println("Node is empty");
        }
    }   
}